### DISCUSSION TITLE
Handling migrations using file hashes

### DISCUSSION THREAD

**Discussion by amitgilad3** [sequelize/umzug#485] [Ideas]
First of all great tool :) ,
i was thinking of adding support for storing migration file hashes inside one of the storages(which ever one the user will choose) and this will make it easy to run the migration programmatically every time one of the migration scripts changes.

I would like to hear any thoughts about this from people and if they think its a good feature ??
if so i would be willing to implement it :)

**Comment by mmkal** [comment]
I think hashed migrations are a good idea, but I'm not totally sure it belongs in umzug. It'd be possible to create a storage class which takes file paths and hashes the content. Given umzug allows in memory migrations too, I'm struggling to think of a clean design. Open to suggestions though.

FYI [@slonik/migrator](https://npmjs.com/package/@slonik/migrator) is an opinionated package built on umzug which does hash migrations in its storage. If your db is postgres, you can use it directly, otherwise you could take a look at the implementation.
